### PR TITLE
Load the ASR runtime library

### PR DESCRIPTION
**Pull request by certik**
This PR makes the following patch possible in Minpack:
```diff
--- a/examples/example_hybrd.f90
+++ b/examples/example_hybrd.f90
@@ -41,10 +41,6 @@ program example_hybrd
     factor = 1.0d2
     nprint = 0
 
-    ! FIXME: This is to trigger LFortran to load the runtime library for this function,
-    ! which we need to link
-    print *, dabs(3.0_8)
-
     call hybrd(fcn, n, x, fvec, xtol, maxfev, ml, mu, epsfcn, diag, &
                mode, factor, nprint, info, nfev, fjac, ldfjac, &
                r, lr, qtf, wa1, wa2, wa3, wa4)
```

### CODE REVIEW

**Comment by certik** [comment]
The Windows error is caused by https://github.com/lfortran/lfortran/issues/1097.

**Comment by certik** [comment]
@czgdp1807 what would you recommend with the design here? I want to get something in rather quickly, and improve upon later, but not sure how to best organize things.

I want to setup some CI tests for this as well, that would fail without this change.

**Comment by czgdp1807** [comment]
Looks fine to me for now. 

> I want to setup some CI tests for this as well, that would fail without this change.

I would be able to comment better once such a test is added.

**Comment by certik** [comment]
TODO:

* [ ] use minpack to create a minimal test in our integration_tests that fails and this PR fixes it

**Comment by certik** [comment]
@Pranavchiku do you want to give this PR a shot and finish it?

**Comment by Pranavchiku** [comment]
Yeah sure!

**Comment by certik** [comment]
To write a test, create a file that calls "sin", then compile it separately using `--generate-object-code`, then in the main program, call some function from the other file. It should fail to link (missing "sin"). Then this PR should fix it. The main program must be compiled with `--rtlib`.

To test it in cmake, we can either improve the "llvm2" test to do this, or write a new "llvm3" test.

**Comment by Pranavchiku** [comment]
I have created minimal examples that will surely produce error, I just need to figure out how to link 3 files that I am using one is a.f90, b.f90, c.f to compile.
Apologies for taking much time over it.

**Comment by Pranavchiku** [comment]
The added tests fail in the main and this PR fixes it. I tested it in the same way we compile `minpack`. I will change `CMakeLists.txt` as soon as possible.

File: src/lfortran/semantics/ast_to_asr.cpp [hunk 1]
```diff
@@ -66,6 +100,7 @@ Result<ASR::TranslationUnit_t*> ast_to_asr(Allocator &al,
         } else {
             return res.error;
         }
+        if (compiler_options.rtlib) load_rtlib(al, *tu, compiler_options);
```
Comment (czgdp1807) [inline src/lfortran/semantics/ast_to_asr.cpp:103]: How are we testing this option in CI? I guess we need to pass it via command line options in `integration_tests/CMakeLists.txt` in order to test it. Please follow the existing patterns and add some support for testing this option in `integration_tests/CMakeLists.txt`.

**Review by czgdp1807 — CHANGES_REQUESTED** [review]

File: src/lfortran/semantics/ast_to_asr.cpp (hunk 1 above)
Comment (Pranavchiku) [reply src/lfortran/semantics/ast_to_asr.cpp:103]: Yeah, sure!

**Comment by Pranavchiku** [comment]
@certik @czgdp1807 I added provision to test it in CI, kindly have a look and comment on the correctness of it.

File: integration_tests/CMakeLists.txt [hunk 2]
```diff
@@ -89,6 +89,13 @@ macro(RUN)
             target_compile_options(${name} PUBLIC --generate-object-code)
             target_link_options(${name} PUBLIC  --generate-object-code)
             add_test(${name} ${CURRENT_BINARY_DIR}/${name})
+        elseif (LFORTRAN_BACKEND STREQUAL "llvm3")
```
Comment (czgdp1807) [inline integration_tests/CMakeLists.txt:92]: Some better name like, `llvm_rtlib`?

**Review by czgdp1807 — APPROVED** [review]

File: src/lfortran/semantics/ast_to_asr.cpp [hunk 3]
```diff
@@ -66,6 +100,7 @@ Result<ASR::TranslationUnit_t*> ast_to_asr(Allocator &al,
         } else {
             return res.error;
         }
+        // if (compiler_options.rtlib) load_rtlib(al, *tu, compiler_options);
```
Comment (czgdp1807) [inline src/lfortran/semantics/ast_to_asr.cpp:103]:
```suggestion
        if (compiler_options.rtlib) load_rtlib(al, *tu, compiler_options);
```

File: src/lfortran/semantics/ast_to_asr.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (certik) [reply src/lfortran/semantics/ast_to_asr.cpp:103]: It fails like this:
```
[100%] Linking Fortran executable minpack_01
Undefined symbols for architecture arm64:
  "___module_lfortran_intrinsic_math2_dabs", referenced from:
      _enorm1 in minpack_01_func.f90.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
The command 'clang -o minpack_01 CMakeFiles/minpack_01.dir/minpack_01.f90.o CMakeFiles/minpack_01.dir/minpack_01_module.f90.o CMakeFiles/minpack_01.dir/minpack_01_func.f90.o  -L"/Users/ondrej/repos/lfortran/lfortran/src/bin/../runtime" -Wl,-rpath,"/Users/ondrej/repos/lfortran/lfortran/src/bin/../runtime" -llfortran_runtime -lm' failed.
```

[the remark this answers is not in the document]
Comment (certik) [reply src/lfortran/semantics/ast_to_asr.cpp:103]: Here it is being tested by the CI: https://github.com/lfortran/lfortran/pull/1174

**Pull request merged**